### DISCUSSION TITLE
Killing a script with ps.pkill also kills salt

### DISCUSSION THREAD

**Discussion by Apteryx0** [saltstack/salt#67481] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
It is not possible to use the ps.pkill command with the full=true option to kill a script without also killing the salt process doing the command. This is illustrated by pgrep, which returns the processes we are interested in but also the salt processes running the command:
```
support@test-cluster-1-master:~$ sudo salt \* ps.pgrep "status_check.py" full=true user=root
worker03:
    None
worker01:
    None
worker02:
    None
test-cluster-1-master:
    - 5104
    - 5105
    - 113618
    - 113629
support@test-cluster-1-master:~$ ps aux | grep "status_check.py"
support    6401  0.0  0.0  13144  1020 pts/1    S+   18:07   0:00 grep --color=auto status_check.py
root     113618  0.0  0.0   4632   820 ?        Ss   18:06   0:00 /bin/sh -c status_check.py check --with-purge 2>&1 | logger -t status_check
root     113629  4.1  0.1 3778936 233456 ?      Sl   18:06   0:04 python /usr/local/sbin/status_check.py check --with-purge
support@test-cluster-1-master:~$ 
```

Additionally the documentation talks about the parameter of both commands being a "pattern", however, looking at the source (https://github.com/saltstack/salt/blob/master/salt/modules/ps.py#L300) just a simple python str "in" operator is used.... so it isn't so much a pattern as a substring.

Both of these issues can be solved by allowing pattern to be a regex, similar to what is used in the ps.psaux function (https://github.com/saltstack/salt/blob/master/salt/modules/ps.py#L745)

There are though other solution that are possible. For example, the ps.psaux command removed salt instances from the results - making pkill/psaux consistent with this would also be welcome.

```
support@test-cluster-1-master:~$ sudo salt \* ps.psaux status_check.py
worker01:
    - status_check.py
    |_
    - 0 occurence(s).
worker03:
    - status_check.py
    |_
    - 0 occurence(s).
test-cluster-1-master:
    - status_check.py
    |_
      - root      94819  0.0  0.0   4632   892 ?        Ss   18:14   0:00 /bin/sh -c status_check.py check --with-purge 2>&1 | logger -t status_check
      - root      94833 34.1  0.1 3777428 231360 ?      Sl   18:14   0:03 python /usr/local/sbin/status_check.py check --with-purge
    - 2 occurence(s).
worker02:
    - status_check.py
    |_
    - 0 occurence(s).
support@test-cluster-1-master:~$ 
```

**Describe the solution you'd like**
Don't include salt commands in the process set *and* support regexs (or update the parameter name from 'pattern' to 'substring')

**Describe alternatives you've considered**
Just supporting pattern being a regex would be sufficient

**Additional context**

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
